demo/api/src/products/product-category.resolver.ts

```diff
@@ -0,0 +1,60 @@
+import { AffectedEntity, CometValidationException, RequiredPermission } from "@comet/cms-api";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Args, ID, Int, Mutation, Resolver } from "@nestjs/graphql";
+
+import { ProductCategory } from "./entities/product-category.entity";
+import { ProductCategoriesService } from "./generated/product-categories.service";
+
+@Resolver(() => ProductCategory)
+@RequiredPermission(["products"], { skipScopeCheck: true })
+export class ProductCategoryResolver {
+    constructor(
+        private readonly entityManager: EntityManager,
+        private readonly productCategoriesService: ProductCategoriesService,
+        @InjectRepository(ProductCategory) private readonly repository: EntityRepository<ProductCategory>,
+    ) {}
+
+    @Mutation(() => Boolean)
+    @AffectedEntity(ProductCategory)
+    async updateProductCategoryPosition(
+        @Args("id", { type: () => ID }) id: string,
+        @Args("position", { type: () => Int }) newPosition: number,
+    ): Promise<boolean> {
+        if (newPosition < 1) {
+            throw new CometValidationException('"position" cannot be smaller than 1');
+        }
+
+        const productCategory = await this.repository.findOneOrFail(id);
+        const { position: oldPosition } = productCategory;
+
+        const count = await this.repository.count({}); // add filter for grouping if necessary
+
+        if (newPosition > count) {
+            throw new CometValidationException(`"position" cannot be greater than ${count}`);
+        }
+
+        await this.entityManager.transactional(async (em) => {
+            if (oldPosition < newPosition) {
+                // Decrement positions between oldPosition (exclusive) and newPosition (inclusive)
+                await this.repository.nativeUpdate(
+                    { position: { $gt: oldPosition, $lte: newPosition } }, // add filter for grouping if necessary
+                    { position: em.createQueryBuilder(ProductCategory).raw("position - 1") },
```

Comment:
you don't need queryBuilder for that, see https://mikro-orm.io/docs/raw-queries

Reply:
raw function is only available in latest mikro-orm, but I was able to simplify it to `em.raw(...)`